numpy/core/_add_newdocs.py

```diff
@@ -3249,9 +3249,13 @@
         * {'|', 'I'} - ignore (no change to byte order)
 
         The default value ('S') results in swapping the current
-        byte order. The code does a case-insensitive check on the first
-        letter of `new_order` for the alternatives above.  For example,
-        any of 'B' or 'b' or 'biggish' are valid to specify big-endian.
+        byte order.
+
+       .. deprecated:: 1.19.0
+
+           Case insensitive and inexact matches are deprecated for new_order.
+           For example, any of 'big', 'biggish' or 'b' in place of 'B' are
```

Comment:
`big` and `little` are consistent with `sys.byteorder`, so I don't think we want to deprecate them.

Reply:
I did this since some of the numpy documentation uses 'B' 'L' etc. for byte order : https://numpy.org/doc/stable/reference/generated/numpy.ndarray.newbyteorder.html?highlight=numpy%20ndarray%20newbyteorder#numpy.ndarray.newbyteorder and I felt deprecating such usage may have more impact on our existing users than in other deprecations. What do you suggest here ? Should we skip byteorder deprecation or just deprecate non exact matches of 'big', 'little' etc.